python/src/aiconfig/eval/lib.py

```diff
@@ -86,23 +86,34 @@ def __str__(self) -> str:
         return f"\nSampleEvaluationParams:\n\t{self.output_sample=}\n\t{self.metric=}"
 
 
-def evaluate(
+async def evaluate(
     evaluation_params_list: Sequence[SampleEvaluationParams[T_InputDatum, T_OutputDatum]],
-) -> Result[DatasetEvaluationResult[T_InputDatum, T_OutputDatum], str]:  # pyright: ignore[fixme, reportInvalidTypeVarUse]
+) -> Result[DatasetEvaluationResult[T_InputDatum, T_OutputDatum], str]:
     results: Sequence[SampleEvaluationResult[T_InputDatum, T_OutputDatum]] = []
 
     for eval_params in evaluation_params_list:
         sample, metric = (
             eval_params.output_sample,
             eval_params.metric,
         )
-        calculate = metric.evaluation_fn
-        res_ = calculate(sample)
-        logger.debug(f"{res_=}")
+
+        async def _calculate() -> MetricValue:
+            return await metric.evaluation_fn(sample)
+
+        def _ok_with_log(res_: Result[MetricValue, str]) -> MetricValue | None:
+            match res_:
+                case Ok(res):
+                    return res
+                case Err(e):
+                    LOGGER.error(f"Error evaluating {eval_params=}: {e}")
+                    return None
+
+        # TODO: figure out the right timeout
+        res_ = await cu.run_thunk_safe(_calculate(), timeout=1)
```

Comment:
is this 1 second? Would be good to add a quick comment, and ideally have timeout be passed in to the function